be/src/vec/functions/function_compress.cpp

```diff
@@ -0,0 +1,242 @@
+            for (size_t i = 0; i < 4; i++) {
+                unsigned char byte = (length >> (i * 8)) & 0xFF;
+                col_data[idx + 2 + i * 2] = hex_itoc[byte >> 4]; // higher four
+                col_data[idx + 3 + i * 2] = hex_itoc[byte & 0x0F];
+            }
+            idx += 10;
+
+            col_data.resize(col_data.size() + 2 * compressed_str.size());
+
+            unsigned char* src = compressed_str.data();
+            for (size_t i = 0; i < compressed_str.size(); i++) {
+                col_data[idx] = hex_itoc[(*src >> 4) & 0x0F];
+                col_data[idx + 1] = hex_itoc[(*src & 0x0F)];
+                idx += 2;
+                src++;
+            }
+            col_offset[row] = col_offset[row - 1] + 10 + compressed_str.size() * 2;
+        }
+
+        block.replace_by_position(result, std::move(result_column));
+        return Status::OK();
+    }
+};
+
+class FunctionUncompress : public IFunction {
+    string hexadecimal = "0123456789ABCDEF";
+    std::map<char, int> hex_ctoi = {
+            {'0', 0},  {'1', 1},  {'2', 2},  {'3', 3},  {'4', 4},  {'5', 5},  {'6', 6},  {'7', 7},
+            {'8', 8},  {'9', 9},  {'A', 10}, {'B', 11}, {'C', 12}, {'D', 13}, {'E', 14}, {'F', 15},
+            {'a', 10}, {'b', 11}, {'c', 12}, {'d', 13}, {'e', 14}, {'f', 15}};
+
+public:
+    static constexpr auto name = "uncompress";
+    static FunctionPtr create() { return std::make_shared<FunctionUncompress>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 1; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return make_nullable(std::make_shared<DataTypeString>());
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        uint32_t result, size_t input_rows_count) const override {
+        // Get the compression algorithm object
+        BlockCompressionCodec* compression_codec;
+        RETURN_IF_ERROR(get_block_compression_codec(segment_v2::CompressionTypePB::ZLIB,
+                                                    &compression_codec));
+
+        const auto& arg_column =
+                assert_cast<const ColumnString&>(*block.get_by_position(arguments[0]).column);
+
+        auto& arg_data = arg_column.get_chars();
+        auto& arg_offset = arg_column.get_offsets();
+        const char* arg_begin = reinterpret_cast<const char*>(arg_data.data());
+
+        auto result_column = ColumnString::create();
+        auto& col_data = result_column->get_chars();
+        auto& col_offset = result_column->get_offsets();
+        col_offset.resize(input_rows_count);
+
+        auto null_column = ColumnUInt8::create(input_rows_count);
+        auto& null_map = null_column->get_data();
+
+        std::string uncompressed;
+        Slice data;
+        Slice uncompressed_slice;
+        for (size_t row = 0; row < input_rows_count; row++) {
```

Comment:
I think we can add reserve for `col_data` first, because resize `input_row_count` times is expensive. reserve with a big enough value is useful.